test/resource_test.rb

```diff
@@ -323,4 +494,56 @@ def destroy_resources
     end
     @resources = []
   end
+
+  # Utility function to test with multiple processes
+  # In particular, this is necessary to ensure handling of unique PIDs is correct,
+  # which is key to functionality that is utilized like SEM_UNDO
+  # Active workers are accumulated in the instance variable @workers,
+  # and workers must be cleaned up between tests by the teardown script
+  # An exit value of 100 is to keep track of timeouts, 0 for success.
+  def fork_workers(count:, resource: :testing, quota: nil, tickets: nil, timeout: 0.1, wait_for_timeout: false)
```

Comment:
Some of the existing tests use `Thread.start` and not fork, we need to settle on one pattern I think unless we're testing specific edge-cases. This somewhat relates to my comment about this not working for threads. I see you already consolidated some of the tests, which is great—but I think we need to do it for the thread tests too.. maybe?